### PR TITLE
Add update API for search contexts

### PR DESCRIPTION
**Pull request by camdencheek**
Does not include an integration test because I think we could expand those in general, and it seems like that should go in a separate PR.

### CODE REVIEW

File: internal/database/search_contexts.go [hunk 1]
```diff
@@ -139,7 +139,10 @@ func (s *SearchContextsStore) listSearchContexts(ctx context.Context, cond *sqlf
 	if err != nil {
 		return nil, err
 	}
-	rows, err := s.Query(ctx, sqlf.Sprintf(listSearchContextsFmtStr, permissionsCond, cond, limit))
+	q := sqlf.Sprintf(listSearchContextsFmtStr, permissionsCond, cond, limit)
+	println(q.Query(sqlf.PostgresBindVar))
```
Comment (novoselrok) [inline internal/database/search_contexts.go:143]: Is this still necessary? 😄

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -603,6 +603,20 @@ type Mutation {
     (experimental) Delete search context.
     """
     deleteSearchContext(id: ID!): EmptyResponse!
+
+    """
+    (experimental) Update search context.
+    """
+    updateSearchContext(
```
Comment (novoselrok) [inline cmd/frontend/graphqlbackend/schema.graphql:610]: I think you're missing an `id: ID!` parameter here, because there is one listed in the `updateSearchContextArgs` struct.

File: internal/gqltestutil/search_context.go [hunk 3]
```diff
@@ -1,6 +1,8 @@
 package gqltestutil
 
 import (
+	"fmt"
```
Comment (novoselrok) [inline internal/gqltestutil/search_context.go:4]: Are you still planning to add integration tests in this PR? If not, could you move these changes to a separate PR/branch for when they will be used, thanks 🙂

**Review by novoselrok — COMMENTED** [review]
Looks good, just some minor comments to resolve.

File: internal/database/search_contexts.go [hunk 4]
```diff
@@ -257,6 +260,35 @@ func (s *SearchContextsStore) CreateSearchContextWithRepositoryRevisions(ctx con
 	return createdSearchContext, nil
 }
 
+const updateSearchContextFmtStr = `
+UPDATE search_contexts
+SET
+	name = %s,
```
Comment (novoselrok) [inline internal/database/search_contexts.go:266]: Also, don't forget to add `updated_at = now()` to the update query.

**Comment by camdencheek** [comment]
D'oh. Thanks for catching these. That's what I get for making PRs while tired 😅

File: internal/gqltestutil/search_context.go (hunk 3 above)
Comment (camdencheek) [reply internal/gqltestutil/search_context.go:4]: Planning on a separate PR. Would probably be a good thing to pair on -- was getting really confused trying to use the graphql API to get contexts.

File: internal/database/search_contexts.go (hunk 1 above)
Comment (camdencheek) [reply internal/database/search_contexts.go:143]: 🤦

File: cmd/frontend/graphqlbackend/schema.graphql (hunk 2 above)
Comment (camdencheek) [reply cmd/frontend/graphqlbackend/schema.graphql:610]: well...that might explain the graphql issues I was having

**Comment by camdencheek** [comment]
@novoselrok Sorry for the bad first PR. It was not in the state I thought I left it. I made the changes you suggested, along with a few other small things. Would you mind giving it another full lookover?

**Review by novoselrok — APPROVED** [review]

**Comment by novoselrok** [comment]
> @novoselrok Sorry for the bad first PR. It was not in the state I thought I left it. I made the changes you suggested, along with a few other small things. Would you mind giving it another full lookover?

No worries, looks good now 😄
